### PR TITLE
[TaskCenter][7/n] Migration of worker crate

### PR DESCRIPTION
**Pull request by AhmedSoliman**
- Removes explicit task-center/metadata from worker
- Migrates worker tests to restate_worker
---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/2355).
* __->__ #2355 (2 commits)
* #2352
* #2344
* #2343
* #2339

### CODE REVIEW

File: crates/worker/src/partition/shuffle.rs [hunk 1]
```diff
@@ -70,13 +69,14 @@ fn create_header(
     seq_number: MessageIndex,
     shuffle_metadata: &ShuffleMetadata,
 ) -> Header {
+    let my_node_id = Metadata::with_current(|m| m.my_node_id());
```
Comment (tillrohrmann) [inline crates/worker/src/partition/shuffle.rs:72]: Could this be `my_node_id()`?

File: crates/worker/src/partition_processor_manager/mod.rs [hunk 2]
```diff
@@ -466,41 +458,33 @@ impl PartitionProcessorManager {
...
                 let result = runtime_task_handle.await;
                 AsynchronousEvent {
                     partition_id,
                     inner: EventKind::Stopped(result),
                 }
-            })
-            .await
-        });
+            }
+            .in_current_tc(),
```
Comment (tillrohrmann) [inline crates/worker/src/partition_processor_manager/mod.rs:469]: This is a lot nicer than before 🤩

**Review by tillrohrmann — APPROVED** [review]
Thanks a lot for cleaning up our code base by improving the `TaskCenter` API @AhmedSoliman. The code looks now a lot nicer than before :-) Really well done!

The only thing I'm wondering now is what did happen to chapter 6/n 🤔 Might be remembered as the lost chapter 😉

**Comment by AhmedSoliman** [comment]
[6/n] is a commit inside this PR :)

**Pull request merged**